docs/troubleshooting-guide/sandboxing.md

```diff
@@ -0,0 +1,56 @@
+# Sandboxing
+
+- Sandboxing means "giving some people access to only some data"
+- Can be set up [graphically][sandboxing-your-data]
+- Metabase runs a query that filters rows and/or selects a subset of columns
+- The user's query then runs on that
+
+So instead of:
+
+```
+select * from orders
+where price > 100.00;
+```
+
+we create a [common table expression][cte] as a replacement starting point for the query:
```

Comment:
more than a CTE, we do a prepared statement

SELECT ... FROM (SELECT * FROM xxx WHERE field_we_use_for_sandbox = $1) "source"
parameters: $1 = 'value'

Reply:
OK, I'm going to need to ask you about this...

Reply:
Several databases (versions) did not support CTEs, when Sandboxing was "invented", but sub-selects were supported by all, so that was chosen.
So your example would generate something like this:
```
SELECT
  "source"."other_columns" AS "other_columns",
  "source"."price" AS "price",
  "source"."customer_id" AS "customer_id" 
FROM
  (
    SELECT
      "orders"."other_columns" AS "other_columns",
      "orders"."price" AS "price",
      "orders"."customer_id" AS "customer_id" 
    FROM
      "orders" 
    WHERE
      "orders"."customer_id" = {{user_id}}
  )
  "source" 
WHERE
  "source"."price" > 100 LIMIT 2000
```